src/main/java/org/opentripplanner/routing/graph/Graph.java

```diff
@@ -993,6 +995,50 @@ public MultiModalStation getMultiModalStation(FeedScopedId id) {
         return stationById.values();
     }
 
+    public StopLocation getStopLocationById(FeedScopedId id) {
+        var stop = index.getStopForId(id);
+        if (stop != null) {
+            return stop;
+        }
+
+        return getAllFlexStopsFlat()
+                .stream()
+                .filter(stopLocation -> stopLocation.getId().equals(id))
+                .findAny()
+                .orElse(null);
+    }
+
+    public StopCollection getStopCollectionById(FeedScopedId id) {
+        var station = stationById.get(id);
+        if (station != null) {
+            return station;
+        }
+
+        var groupOfStations = groupOfStationsById.get(id);
+        if (groupOfStations != null) {
+            return groupOfStations;
+        }
+
+        return multiModalStationById.get(id);
+    }
+
+    public Stream<StopLocation> getAllStopLocations() {
+        return Stream.concat(
+                index.getAllStops().stream(),
+                getAllFlexStopsFlat().stream()
+        );
+    }
+
+    public Stream<StopCollection> getAllStopCollections() {
+        return Stream.concat(
+                stationById.values().stream(),
+                Stream.concat(
+                        groupOfStationsById.values().stream(),
+                        multiModalStationById.values().stream()
+                )
+        );
+    }
```

Comment:
Can you document these methods? The last two methods are only used by the LuceneIndex. They might be re-useable, but they are potentially harmful if used in a API query. 
 
 I want to make a LocationService witch encapsulate the all stop/station and so on. I will try to remember that we probably should have two interfaces to this service - one for routing (backed up by indexes) and one with utility methods, like the last too.